appserver/common/container-common/src/main/java/com/sun/enterprise/container/common/impl/managedbean/ManagedBeanManagerImpl.java

```diff
@@ -401,21 +391,11 @@ public void unloadManagedBeans(Application app) {
     }
 
     private boolean bundleEligible(BundleDescriptor bundle) {
-
-        boolean eligible = false;
-
-        if( processType.isServer() ) {
-
-            eligible = (bundle instanceof WebBundleDescriptor) ||
-                (bundle instanceof EjbBundleDescriptor);
-
-        } else if( processType == ProcessType.ACC ) {
-            eligible = (bundle instanceof ApplicationClientDescriptor);
-        }
-
-        return eligible;
+        return processType.isServer() && (bundle instanceof WebBundleDescriptor || bundle instanceof EjbBundleDescriptor)
```

Comment:
I would rather use something in between ... the expression is too complicated to read, so I would split it to more blocks with returns:
```
        if (processType.isServer() ) {
            return (bundle instanceof WebBundleDescriptor) || (bundle instanceof EjbBundleDescriptor);
        } 
        if (processType == ProcessType.ACC) {
            return bundle instanceof ApplicationClientDescriptor;
        }
        return false;
```
Also it looks weird that one branch calls isServer method and second compares type.
Another problem is using && and || on the same level without braces `A && B || C` - for newbies and sleepy developers it is hard to read, they make mistakes.